client/src/main/java/org/threadly/litesockets/client/http/HTTPStreamClient.java

```diff
@@ -166,7 +166,7 @@ public void setRequestResponseHeaders(HTTPRequest httpRequest, HTTPResponse http
       slfResponse.setFailure(new IllegalStateException("New request came in!"));
     }
     currentHttpRequest = request;
-    slfResponse = new SettableListenableFuture<HTTPResponse>();
+    slfResponse = new SettableListenableFuture<HTTPResponse>(false);
```

Comment:
A quick look makes it seem like this future (through use of requestCB) will complete on the client thread (expect for the failure above).

What do you think about allowing you to queue / pipeline multiple requests by changing the `setFailure` to:
```
  return slfResponse = slfResponse.map((ignored) -> writeRequest(request))
```

Then we could change this to be a ClientSettableListenableFuture, and allow possible optimizations there.  That would be my primary motivation, but the more I think about this, I am unsure about how we could easily handle delaying writes after this until they can occur after the request is sent.

Reply:
I guess I dont fully understand how that works.  Right now we dont allow queueing of requests, and for alot of cases it would not make sense.  Most streaming http connections cant be reused to even serve another request.  Though you could use this for many non-streaming requests I think the response bodies on the Reader would get pretty difficult to tie back to certain requests.

Actually looking at it closer it looks like this cant even handle multi-requests since we will call close on the client right after we fully process a request.  I do think we might need to rework this as I dont feel like its a very clean implementation, might need to make a Streaming client and a multiRequest client that just work differently.

Reply:
Yeah, I am fine shelving it for now, I don't exactly know why we need multiple clients vs a single client that just supports multiple modes of usage.  But could be something for improvement later.